lib/product.rb

```diff
@@ -0,0 +1,17 @@
+class Product
+  attr_reader :category, :name, :unit_price, :quantity
+
+  def initialize(category, name, unit_price, quantity)
+    @category = category
+    @name = name
+    @unit_price = unit_price
+    @quantity = quantity.to_i
+  end
+
+  # returns the sum of the unit price multiplied by the quantity
+  # no argument -> float
+  def total_price
+    unit_price * quantity
+  end
+
```

Comment:
Oh no!  you're missing two methods from iteration 1 😢

Reply:
uh oh lol that's totally on me I completely missed those two methods 🥲